x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -184,59 +185,98 @@ export class ExecutionHandler<
+                kibanaBaseUrl: this.taskRunnerContext.kibanaBaseUrl,
+                alertParams: this.rule.params,
+                actionParams: action.params,
+                ruleUrl: this.buildRuleUrl(spaceId),
+              }),
             }),
-          }),
-        };
-
-        await this.actionRunOrAddToBulk({
-          enqueueOptions: this.getEnqueueOptions(actionToRun),
-          bulkActions,
-        });
-
-        logActions.push({
-          id: action.id,
-          typeId: action.actionTypeId,
-          alertId: alert.getId(),
-          alertGroup: action.group,
-        });
-
-        if (this.isRecoveredAlert(actionGroup)) {
-          alert.scheduleActions(action.group as ActionGroupIds);
-        } else {
-          if (action.frequency?.throttle) {
-            alert.updateLastScheduledActions(
-              action.group as ActionGroupIds,
-              this.generateActionHash(action)
-            );
+          };
+
+          await this.actionRunOrAddToBulk({
+            enqueueOptions: this.getEnqueueOptions(actionToRun),
+            bulkActions,
+          });
+
+          logActions.push({
+            id: action.id,
+            typeId: action.actionTypeId,
+            alertId: alert.getId(),
+            alertGroup: action.group,
+          });
+
+          if (this.isRecoveredAlert(actionGroup)) {
+            alert.scheduleActions(action.group as ActionGroupIds);
           } else {
-            alert.updateLastScheduledActions(action.group as ActionGroupIds);
+            if (this.isActionOnThrottleInterval(action)) {
+              alert.updateLastScheduledActions(
+                action.group as ActionGroupIds,
+                this.generateActionHash(action)
+              );
+            } else {
+              alert.updateLastScheduledActions(action.group as ActionGroupIds);
+            }
+            alert.unscheduleActions();
+          }
+        } else {
+          const summarizedAlerts = await this.getSummarizedAlerts({ action, spaceId, ruleId });
+          const actionToRun = {
+            ...action,
+            params: injectActionParams({
+              ruleId,
+              spaceId,
+              actionTypeId,
+              actionParams: transformSummaryActionParams({
+                alerts: summarizedAlerts,
+                rule: this.rule,
+                ruleTypeId: this.ruleType.id,
+                actionId: action.id,
+                actionParams: action.params,
+                spaceId,
+                actionsPlugin,
+                actionTypeId,
+                kibanaBaseUrl: this.taskRunnerContext.kibanaBaseUrl,
+                ruleUrl: this.buildRuleUrl(spaceId),
+              }),
+            }),
+          };
+
+          await this.actionRunOrAddToBulk({
+            enqueueOptions: this.getEnqueueOptions(actionToRun),
+            bulkActions,
+          });
+
+          if (this.isActionOnThrottleInterval(action)) {
```

Comment:
**question:** should this replace the `} else {` code above?

```
} else if (this.isActionOnThrottleInterval(action)) {
... this code block ...
}
```

Reply:
Nope, (if i got the correct `} else {` :) ) it could be a `Summary of alerts per rule run` 
this check is for adding the summary action trigger date to the task instance only for the `Summary of alerts on a custom interval`